File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationBottomOfParent.java
package com.github.florent37.expectanim.core.position;

import android.view.View;
import android.view.ViewParent;

/**
 * Created by florentchampigny on 17/02/2017.
 */

public class PositionAnimExpectationBottomOfParent extends PositionAnimExpectation {

    public PositionAnimExpectationBottomOfParent() {
        setForPositionY(true);
    }

    @Override
    public Float getCalculatedValueX(View viewToMove) {
        return null;
    }

    @Override
    public Float getCalculatedValueY(View viewToMove) {
        final ViewParent viewParent = viewToMove.getParent();
        if((viewParent instanceof View)){
            final View parentView = (View)viewParent;
            return parentView.getHeight() - getMargin(viewToMove) - viewCalculator.finalHeightOfView(viewToMove);
        }
        return null;
    }
}

File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationBottomOfParent.kt
package com.github.florent37.expectanim.core.position

import android.view.View
import android.view.ViewParent

class PositionAnimExpectationBottomOfParent : PositionAnimExpectation() {

    init {
        isForPositionY = true
    }

    override fun getCalculatedValueX(viewToMove: View): Float? {
        return null
    }

    override fun getCalculatedValueY(viewToMove: View): Float? {
        val viewParent = viewToMove.parent
        if (viewParent is View) {
            val parentView = viewParent as View
            return parentView.height.toFloat() - getMargin(viewToMove) - viewCalculator!!.finalHeightOfView(viewToMove)
        }
        return null
    }
}
